nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/GetHDFSFileInfo.java

```diff
@@ -255,6 +270,30 @@ public void onPropertyModified(PropertyDescriptor descriptor, String oldValue, S
         req = null;
     }
 
+    @Override
+    protected Collection<ValidationResult> customValidate(ValidationContext validationContext) {
+        final List<ValidationResult> validationResults = new ArrayList<>(super.customValidate(validationContext));
+
+        String destination = validationContext.getProperty(DESTINATION).getValue();
+        String grouping = validationContext.getProperty(GROUPING).getValue();
+        String batchSize = validationContext.getProperty(BATCH_SIZE).getValue();
+
+        if (
+            (!DESTINATION_CONTENT.getValue().equals(destination) || !GROUP_NONE.getValue().equals(grouping))
+                && !isEmpty(batchSize)
```

Comment:
If we avoid allowing Batch Size to be set to an empty string, as recommended above, we can instead just check `validationContext.getProperty(BATCH_SIZE).isSet()` here. I think this reads more cleanly, and is more inline with the way most processors are written.